src/lib.rs

```diff
@@ -258,6 +258,19 @@ impl<T: Runtime> Client<T> {
         &self.metadata
     }
 
+    /// Fetch the value under an unhashed storage key
+    pub async fn fetch_unhashed<V: Decode>(
+        &self,
+        key: StorageKey,
+        hash: Option<T::Hash>,
+    ) -> Result<Option<V>, Error> {
+        if let Some(data) = self.rpc.storage(&key, hash).await? {
+            Ok(Some(Decode::decode(&mut &data.0[..])?))
+        } else {
+            Ok(None)
+        }
+    }
```

Comment:
One more thing: could you modify `fetch` below :point_down: to call this method just to reduce the duplication.